### PR TITLE
Add proposal for the users.list filter on term

### PR DESCRIPTION
**Pull request by WouterSioen**
Add a filter on the users.list endpoint.

It will filter on first name, last name and email.

### CODE REVIEW

**Comment by marcwestermann** [comment]
Hey Wouter, thanks for picking this up! The name (first name and last name) is the most interesting filter for us. We are trying to filter the users on an API level to select them as attendees on the new calendar. Maybe, function (position) would also be nice to have, but not required.
image
@duivvv , what do you think? :)

**Comment by geoffreydhuyvetters** [comment]
I think function is nice to have but not essential.
The combination `company` / `function` makes sense in the context of a contact (> 'ceo madewithlove' for example)

File: src/01-general/users.apib [hunk 1]
```diff
@@ -48,6 +48,7 @@ Get a list of all users.
     + Attributes (object)
         + filter (object, optional)
             + ids: `cb8da52a-ce89-4bf6-8f7e-8ee6cb85e3b5`,`f8a57a6f-dd1e-41a3-b8d3-428663f1d09e` (array[string], optional)
+            + term: `John` (string, optional) - Filters on username, real name and email address
```
Comment (dietrichm) [inline src/01-general/users.apib]: B: there is no notion of "real name" in the return data for users. The first and last name is returned as a result from splitting on the first space.

```suggestion
            + term: `John` (string, optional) - Filters on username, first_name, last_name and email
```

Also the username still needs to be added to the user info.

**Review by dietrichm — CHANGES_REQUESTED** [review]

**Review by dietrichm — DISMISSED** [review]
PP: username still should be added to the return data.

**Review by mrtus — DISMISSED** [review]

File: apiary.apib [hunk 2]
```diff
@@ -466,6 +466,7 @@ Get a list of all users.
     + Attributes (object)
         + filter (object, optional)
             + ids: `cb8da52a-ce89-4bf6-8f7e-8ee6cb85e3b5`,`f8a57a6f-dd1e-41a3-b8d3-428663f1d09e` (array[string], optional)
+            + term: `John` (string, optional) - Filters on username, first_name, last_name and email
```
Comment (mrtus) [inline apiary.apib]: Q: Should we leave out username for now? I don't think it has any use for now.

**Review by dietrichm — APPROVED** [review]

**Review by JordyDeGroeve — APPROVED** [review]

**Review by mrtus — APPROVED** [review]

**Review by marcwestermann — APPROVED** [review]

**Review by tiagobastos — APPROVED** [review]

**Review by jenssegers — APPROVED** [review]

**Pull request merged**